extensions/eclipse-che-theia-plugin-ext/src/browser/che-workspace-main.ts

```diff
@@ -39,4 +44,26 @@ export class CheWorkspaceMainImpl implements CheWorkspaceMain {
         return await this.workspaceService.updateWorkspace(workspaceId, workspace);
     }
 
+    async $restartWorkspace(machineToken: string, prompt: boolean = true): Promise<boolean> {
+        if (prompt) {
+            const confirm = new ConfirmDialog({
+                title: 'Restart Workspace',
+                msg: 'Are you sure you want to restart your workspace?',
```

Comment:
I imagine a real use case when suddenly the user is asked such a question...
Doesn't it sound a little bit scary? :smile: 

Maybe add a bit more details here...?
![image](https://user-images.githubusercontent.com/1636395/97721747-8282da00-1ad2-11eb-86eb-3dd983ab88b7.png)
WDYT?

Reply:
maybe it makes sense to allow a plugin to provide the message for `Restart Workspace` dialog

Reply:
so one more parameter for the method